tests/functional/test_useragent.py

```diff
@@ -0,0 +1,330 @@
+# Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import logging
+from contextlib import contextmanager
+from datetime import datetime
+from itertools import product
+
+import pytest
+
+from botocore import __version__ as botocore_version
+from botocore.config import Config
+from botocore.stub import Stubber
+
+
+@contextmanager
+def uacap_client(client):
+    """Contextmanager yielding client object that captures the last UA header
+
+    Usage:
+
+        with uacap_client(myclient) as cap_client:
+            myclient.operation()
+        print(cap_client.captured_ua_string)
+    """
+    client.captured_ua_string = None
```

Comment:
Would there be any value to add or replace this with a list that captures sequential `ua_string`'s?

Reply:
I ended up replacing this entire fixture in eb7ded6 to address another comment above. To answer the question though: I didn't see a reason to do more than one request in a single test because each request's UA is independent of the others.